### ISSUE TITLE
WHMCS 8.1.3, 8.2.0 8.2.1 not working 

### ISSUE THREAD

**Issue by HackersThirdEyeofficial** [vultr/whmcs-vultr#115]
This Module is not working anymore on whmcs 8.1.3, 8.2.0 8.2.1, 

when the user purchases the service VM created in the backend without any issue! but the user is not able to manage their server, always Show you have to create VM when it's already created

Bug.2 Same User can create an unlimited server from the same package in the backend.

Anyone Who can update this module?

**Comment by habebaessam12** [comment]
have you tries whmcs 7.9?

**Comment by habebaessam12** [comment]
tried*

**Comment by nhaminul** [comment]
Same here, Version 8.2.1 does not work. Please solve it.

**Comment by 1shohan** [comment]
Anyone here who can solve it or solve the problem?

**Comment by shahidmallaofficial** [comment]
No

**Comment by shahidmallaofficial** [comment]
i think we should pay for it

**Comment by nhaminul** [comment]
Developer bro please reply here.

**Comment by wpexpertinbd** [comment]
Hello guys, after few months later I comes to this page & found some of you are facing the issues. I found it's actually working fine. Vultr moudle  actually working as you mention at first comment. What is full process to make it working or how it's working that are below:
1. Your client placed the order fine
2. You can accept order manually or automatically, which one use that does not matter , you should not use default vps/dedicated server welcome email. You need to just mention your client that his VPS is activated kindly login to your panel & deploy your OS
3. If you want to deploy the vps & want to send the real IP, user id & password to your client then after accepting order, login as client & deploy the VPS as per your client need, get all login details & save them at whmcs product details page  of your client
4. Finally you can resend the VPS welcome email
5. That means you need to deploy the VPS from client end or your client need to deploy it, that's the matter nothing else

If anyone still facing any issue kindly let me know!

**Comment by shahidmallaofficial** [comment]
> Hello guys, after few months later I comes to this page & found some of you are facing the issues. I found it's actually working fine. Vultr moudle actually working as you mention at first comment. What is full process to make it working or how it's working that are below:
> 
> 1. Your client placed the order fine
> 2. You can accept order manually or automatically, which one use that does not matter , you should not use default vps/dedicated server welcome email. You need to just mention your client that his VPS is activated kindly login to your panel & deploy your OS
> 3. If you want to deploy the vps & want to send the real IP, user id & password to your client then after accepting order, login as client & deploy the VPS as per your client need, get all login details & save them at whmcs product details page  of your client
> 4. Finally you can resend the VPS welcome email
> 5. That means you need to deploy the VPS from client end or your client need to deploy it, that's the matter nothing else. Don't deploy the VPS from admin panel, just do it from client area after activating the order.
> 
> My Setup: WHMCS: 8.2.1 PHP: 7.4.23 Vultr Module
> 
> If anyone still facing any issue kindly let me know! ![vultr-working-1](https://user-images.githubusercontent.com/26407345/135751075-b1615d5f-69cf-477d-b210-718f8eb3a088.png)

there is no such option when we add the module and assign products because it shows an error when users create the account.  may be you have modified code

**Comment by wpexpertinbd** [comment]
@shahidmallaofficial , which option you are talking about. You need to follow this module installation guide carefully. Like module installation, custom field/Configurable option creating e.t.c. Then you can place order manually or your client can place order from client area. You can accept the order as like as others product, but you or your client need to create the VM from client area, don't try to create the VM from whmcs admin panel. That's all. If you still not able to make it working then you need to hire someone to make it working, because you really don't understand what you need to do or what I am saying here. Thanks.

**Comment by limsoke** [comment]
i got the same, but added Virtual machine ID in admin area user product fixed for me.

**Comment by meramsey** [comment]
Advise looking into this maintained product which is what I switched too.
https://www.modulesgarden.com/products/whmcs/vultr-vps

 This made it finally possible to transition the existing clients in place by adding a new customfield to all existing Vultr Products "instanceId|Instance ID" text admin only. I then went and populated all existing orders with their V2 API instance id which you can get from visiting the server in the my.vultr.com dashboard and noting the string from url or via the V1/V2 list servers/instances API "v2_id": field. I then was able to just switch the products Module from Vultr to Vultr VPS and all the existing server's connections remain working.

**Comment by shahidmallaofficial** [comment]
> Advise looking into this maintained product which is what I switched too. https://www.modulesgarden.com/products/whmcs/vultr-vps
> 
> This made it finally possible to transition the existing clients in place by adding a new customfield to all existing Vultr Products "instanceId|Instance ID" text admin only. I then went and populated all existing orders with their V2 API instance id which you can get from visiting the server in the my.vultr.com dashboard and noting the string from url or via the V1/V2 list servers/instances API "v2_id": field. I then was able to just switch the products Module from Vultr to Vultr VPS and all the existing server's connections remain working.

yes it seems nice but expensive per month 13$  now additional cost, i am paying cpanel 200$ + lagom 12$ + whmcs 37$ + servers 700$ + imunify 45*5$ + cloudlinux 10*7$ + Litespeed 24*5$  + plesk 80$ permonth and other things,     

these big tech make our startup companies and individuals like hell  

and I have also closed the business and will try to some amount so we can invest to develop our custom products there is no reason spending much monthly or per client.  

the developers of modulegurden were doing well but after whmcs changed their pricing make all modulegurdn use less because we are almost ready to shift from whmcs! 

i just want to recommend every startup invest more on developing its own custom applications if it's not good at starting!

**Comment by Businezz99** [comment]
> > Advise looking into this maintained product which is what I switched too. https://www.modulesgarden.com/products/whmcs/vultr-vps
> > This made it finally possible to transition the existing clients in place by adding a new customfield to all existing Vultr Products "instanceId|Instance ID" text admin only. I then went and populated all existing orders with their V2 API instance id which you can get from visiting the server in the my.vultr.com dashboard and noting the string from url or via the V1/V2 list servers/instances API "v2_id": field. I then was able to just switch the products Module from Vultr to Vultr VPS and all the existing server's connections remain working.
> 
> yes it seems nice but expensive per month 13$ now additional cost, i am paying cpanel 200$ + lagom 12$ + whmcs 250$ + servers 700$ + imunify 45_5$ + cloudlinux 10_7$ + Litespeed 24*5$ + plesk 80$ permonth and other things,
> 
> these big tech make our startup companies and individuals like hell
> 
> and I have also closed the business and will try to some amount so we can invest to develop our custom products there is no reason spending much monthly or per client.
> 
> the developers of modulegurden were doing well but after whmcs changed their pricing make all modulegurdn use less because we are almost ready to shift from whmcs!
> 
> i just want to recommend every startup invest more on developing its own custom applications if it's not good at starting!

I agree with you. Especially modules garden is charging a lot and indeed, they have useful modules. But you are right, they are killing the small businesses with all increased prices.
